src/coprocessor/codec/convert.rs

```diff
@@ -261,6 +289,92 @@ pub fn convert_float_to_uint(ctx: &mut EvalContext, fval: f64, tp: FieldTypeTp)
     Ok(val as u64)
 }
 
+/// Converts a byte arrays to an u64 in best effort.
+#[inline]
+pub fn convert_bytes_to_uint(ctx: &mut EvalContext, bytes: &[u8], tp: FieldTypeTp) -> Result<u64> {
+    let s = str::from_utf8(bytes)?.trim();
+    let vs = get_valid_int_prefix(ctx, s)?;
+    let val = vs.parse::<u64>();
+    match val {
+        Ok(val) => convert_uint_to_uint(ctx, val, tp),
+        Err(_) => {
+            ctx.handle_overflow(Error::overflow("BIGINT UNSIGNED", &vs))?;
+            let val = integer_unsigned_upper_bound(tp);
+            Ok(val)
+        }
+    }
+}
+
+macro_rules! decimal_as_u64 {
```

Comment:
Why we need use macro here, I think an inline function is ok.
using funciton but not macro is friendly to IDE